_posts/2023-05-11-possearchproducts.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: post
+title: "[CVE-2023-30192] Improper neutralization of SQL parameter in Posthemes Search Products for PrestaShop"
+categories: modules
+author:
+- Touchweb.fr
+- 202 ecommerce.com
+- Friends-Of-Presta.org
+meta: "CVE,PrestaShop,possearchproducts"
+severity: "critical (9.8)"
+---
+
+In the module "Search Products" (possearchproducts), a guest can perform SQL injection in affected versions.
+
+## Summary
+
+* **CVE ID**: [CVE-2023-30192](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-30192)
+* **Published at**: 2023-05-11
+* **Advisory source**: Friends-Of-Presta.org
+* **Platform**: PrestaShop
+* **Product**: possearchproducts
+* **Impacted release**: <= 1.7 (VERSION'S SCOPE NOT CONFIRMED - AUTHOR NEVER ANSWER)
+* **Product author**: posthemes
+* **Weakness**: [CWE-89](https://cwe.mitre.org/data/definitions/89.html)
+* **Severity**: critical (9.8)
+
+## Description
+
+The method `PosSearch::find()` has sensitive SQL calls that can be executed with a trivial http call and exploited to forge a SQL injection.
+
+The exploit can be used even if the module is not activated.
+
+**WARNING** : This exploit is actively used to deploy webskimmer to massively stole credit cards.
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: low
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: high
+* **Availability**: high
+
+**Vector string**: [CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H)
+
+## Possible malicious usage
+
+* Obtain admin access
+* Remove data on the associated PrestaShop
+* Copy/past datas from sensibles tables to FRONT to exposed tokens and unlock admins's ajax scripts
+* Rewrite SMTP settings to hijacked emails
+
+
+## Proof of concept
+
+
+```bash
+curl -v 'https://preprod.XXX/modules/possearchproducts/SearchProducts.php?s=test&id_category=1;select(0x73656C65637420736C656570283432293B)INTO@a;prepare`b`from@a;execute`b`;--'
+```
+
+## Patch from 1.7
+
+```diff
+--- 1.7/modules/possearchproducts/PosSearch.php
++++ XXX/modules/possearchproducts/PosSearch.php
+...
+WHERE c.`active` = 1
+-        '.($id_category !=  0 ? 'AND c.`id_category` = '.$id_category.'':'').'
++        '.($id_category !=  0 ? 'AND c.`id_category` = '.(int) $id_category.'':'').'
+```
+
+Be warn that there is other sensitives SQL calls inside this module accessible to administrators. Since there is thousand of injection SQL accessible to administrators on the Prestashop's ecosystem, these vulnerabilities are ignored until author provide a patch.
```

Comment:
be warn**ed** non?


Presta**S**hop

Reply:
Le paragraphe avait été repris du fait d'un C/C depuis les autres pos, je l'ai supprimé.